py-polars/polars/internals/frame.py

```diff
@@ -2017,9 +2020,12 @@ def filter(self, predicate: pli.Expr | str | pli.Series | list[bool]) -> DataFra
         └─────┴─────┴─────┘
 
         """
+        if _NUMPY_AVAILABLE and isinstance(predicate, np.ndarray):
+            predicate = pli.Series(predicate)
+
         return (
             self.lazy()
-            .filter(predicate)
+            .filter(predicate)  # type: ignore[arg-type]
             .collect(no_optimization=True, string_cache=False)
         )
```

Comment:
FWIW if we implement it like

```python
        if _NUMPY_AVAILABLE and isinstance(predicate, np.ndarray):
            lazy_predicate = pli.Series(predicate)
        else:
            lazy_predicate = predicate

        return (
            self.lazy()
            .filter(lazy_predicate)
            .collect(no_optimization=True, string_cache=False)
        )
```
we won't need the type ignore

Reply:
Right.. Sorry I read this too late. Was in a hurry to ship the new release. :)